developer_manual/app/advanced/code_signing.rst

```diff
@@ -91,108 +88,60 @@ following content:
       "signature": "U29tZVNpZ25lZERhdGFFeGFtcGxl"
   }
 
-``hashes`` is an array of all files in the folder with their corresponding 
-SHA-512 hashes. ``certificate`` is the certificate used for signing. It has to 
-be issued by the ownCloud Root Authority, and its CN needs to be permitted to 
-perform the required action. The ``signature`` is then a signature of the hashes 
-which can be verified using the certificate.
+``hashes`` is an array of all files in the folder with their corresponding SHA-512 hashes. 
+``certificate`` is the certificate used for signing. 
+It has to be issued by the ownCloud Root Authority, and its CN needs to be permitted to perform the required action. 
+The ``signature`` is then a signature of the hashes which can be verified using the certificate.
 
-Having the certificate bundled within the ``signature.json`` file has the 
-advantage that even if a developer loses their certificate, future updates can 
-still be ensured by having a new certificate issued.
+Having the certificate bundled within the ``signature.json`` file has the advantage that even if a developer loses their certificate, future updates can still be ensured by having a new certificate issued.
 
 How Code Signing Affects Apps in the ownCloud Marketplace
 ---------------------------------------------------------
 
-- Apps which have an ``official`` tag **MUST** be code signed starting with 
-  ownCloud 9.0. Unsigned ``official`` apps won't be installable anymore.
-- Apps which have been signed in a previous release **MUST** be code-signed in 
-  all future releases as well, otherwise the update will be refused.
+- Apps which have an ``official`` tag **MUST** be code signed starting with ownCloud 9.0. Unsigned ``official`` apps won't be installable anymore.
```

Comment:
As in the other comment. Unsigned apps can't be uploaded to the marketplace. They can be installed manually but you will always see a warning: "Integrity check failed"